hypha/apply/projects/views/vendor.py

```diff
@@ -0,0 +1,304 @@
+import json
+
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.core.exceptions import PermissionDenied
+from django.db.models.fields.files import FieldFile
+from django.http import Http404
+from django.shortcuts import get_object_or_404, render
+from django.utils import timezone
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext_lazy as _
+from django.views.generic.detail import DetailView
+from formtools.wizard.views import SessionWizardView
+from wagtail.core.models import Site
+
+from addressfield.fields import ADDRESS_FIELDS_ORDER
+from hypha.apply.projects.models.vendor import VendorFormSettings
+from hypha.apply.utils.storage import PrivateMediaView, PrivateStorage
+
+from ..forms import (
+    CreateVendorFormStep1,
+    CreateVendorFormStep2,
+    CreateVendorFormStep3,
+    CreateVendorFormStep4,
+    CreateVendorFormStep5,
+    CreateVendorFormStep6,
+)
+from ..models import (
+    BankInformation,
+    DueDiligenceDocument,
+    Project,
+    ProjectSettings,
+    Vendor,
+)
+
+
+def show_extra_info_form(wizard):
+    # try to get the cleaned data of step where we ask if extra info is needed
+    cleaned_data = wizard.get_cleaned_data_for_step('extra') or {}
+    # check if the field ``extra_info`` box was checked.
+    return cleaned_data.get('need_extra_info', True)
+
+
+class VendorAccessMixin:
+    def dispatch(self, request, *args, **kwargs):
+        project_settings = ProjectSettings.for_request(request)
+        if not project_settings.vendor_setup_required:
+            raise PermissionDenied
+        is_admin = request.user.is_apply_staff
+        project = self.get_project()
+        is_owner = request.user == project.user
+        if not (is_owner or is_admin):
+            raise PermissionDenied
+        if not project.vendor:
+            raise Http404
```

Comment:
Does this not  make it impossible to create a vendor for a new project since VendorAccessMixin is used for CreateVendorView?